src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -88,23 +88,24 @@ def _update_latest_from_pypi(versions):
     from subprocess import check_output, STDOUT, CalledProcessError
 
     success = False
-    try:
-        cmd = [sys.executable] + '-m pip search azure-cli -vv --disable-pip-version-check --no-cache-dir'.split()
-        logger.debug('Running: %s', cmd)
-        log_output = check_output(cmd, stderr=STDOUT, universal_newlines=True)
-        success = True
-        for line in log_output.splitlines():
-            if not line.startswith(CLI_PACKAGE_NAME):
-                continue
-            comps = line.split()
-            mod = comps[0].replace(COMPONENT_PREFIX, '') or CLI_PACKAGE_NAME
-            version = comps[1].replace('(', '').replace(')', '')
-            try:
-                versions[mod]['pypi'] = version
-            except KeyError:
-                pass
-    except CalledProcessError:
```

Comment:
IMHO, I would keep this try/catch for pip failure even if we've added a fail-fast above. It'll keep the robust behavior before, with the improved fail-fast. I think the counter argument, which I can't really speak to, is that the Python idiom would be to have the exception propagate up instead of returning a bool to indicate that the check was unsuccessful.